command/auto_complete.go

```diff
@@ -0,0 +1,247 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/peak/s5cmd/storage"
+	"github.com/peak/s5cmd/storage/url"
+	"github.com/urfave/cli/v2"
+)
+
+const zsh = `autoload -Uz compinit
+compinit
+
+_s5cmd_cli_zsh_autocomplete() {
+	local -a opts
+	local cur
+	cur=${words[-1]}
+	opts=("${(@f)$(${words[@]:0:#words[@]-1} "${cur}" --generate-bash-completion)}")
+
+	if [[ "${opts[1]}" != "" ]]; then
+	  _describe 'values' opts
+	else
+	  _files
+	fi
+}
+
+compdef _s5cmd_cli_zsh_autocomplete s5cmd
+`
+
+const bash = `# prepare autocompletion suggestions for s5cmd and save them to COMPREPLY array
+_s5cmd_cli_bash_autocomplete() {
+	if [[ "${COMP_WORDS[0]}" != "source" ]]; then
+		COMPREPLY=()
+		local opts cur cmd` +
+	// get current word (cur) and prepare command (cmd)
+	`
+		cur="${COMP_WORDS[COMP_CWORD]}"
+		cmd="${COMP_LINE:0:$COMP_POINT}"` +
+
+	// if we want to complete the second argurment and we didn't started writing
+	// yet then we should pass an empty string as another argument. Otherwise
+	// the white spaces will be discarded and the program will make suggestions
+	// as if it is completing the first argument.
+	// Beware that the we want to pass empty string so we intentionally write
+	// as it is. Fixes of SC2089 and SC2090 are not what we want.
```

Comment:
```suggestion
	// if we want to complete the second argument and we didn't start writing
	// yet then we should pass an empty string as another argument. Otherwise
	// the white spaces will be discarded and the program will make suggestions
	// as if it is completing the first argument.
	// Beware that we want to pass an empty string so we intentionally write
	// as it is. Fixes of SC2089 and SC2090 are not what we want.
```